pkg/kv/kvserver/spanset/spanset_test.go

```diff
@@ -381,16 +388,106 @@ func TestSpanSetWriteImpliesRead(t *testing.T) {
 	ss.AddNonMVCC(SpanReadOnly, roSpan)
 	ss.AddNonMVCC(SpanReadWrite, rwSpan)
 
-	if err := ss.CheckAllowed(SpanReadOnly, roSpan); err != nil {
+	if err := ss.CheckAllowed(SpanReadOnly, TrickySpan(roSpan)); err != nil {
 		t.Errorf("expected to be allowed to read roSpan, error: %+v", err)
 	}
-	if err := ss.CheckAllowed(SpanReadWrite, roSpan); err == nil {
+	if err := ss.CheckAllowed(SpanReadWrite, TrickySpan(roSpan)); err == nil {
 		t.Errorf("expected not to be allowed to write roSpan")
 	}
-	if err := ss.CheckAllowed(SpanReadOnly, rwSpan); err != nil {
+	if err := ss.CheckAllowed(SpanReadOnly, TrickySpan(rwSpan)); err != nil {
 		t.Errorf("expected to be allowed to read rwSpan, error: %+v", err)
 	}
-	if err := ss.CheckAllowed(SpanReadWrite, rwSpan); err != nil {
+	if err := ss.CheckAllowed(SpanReadWrite, TrickySpan(rwSpan)); err != nil {
 		t.Errorf("expected to be allowed to read rwSpan, error: %+v", err)
 	}
 }
+
+// makeSpanHelper accepts strings like: "a-d", and returns a span with
+// startKey = a, and endKey = d. It also accepts `X` which represents nil. For
+// example, "X-d" returns a span with a nil startKey, and endKey = d.
+func makeSpanHelper(t *testing.T, s string) roachpb.Span {
+	parts := strings.Split(s, "-")
+	require.Len(t, parts, 2)
+
+	var start roachpb.Key
+	var end roachpb.Key
+
+	if parts[0] != "X" {
+		start = roachpb.Key(parts[0])
+	}
+
+	if parts[1] != "X" {
+		end = roachpb.Key(parts[1])
+	}
```

Comment:
style nit (feel free to ignore, just noticed the pattern in this PR):

In simple code, vertical sparseness / empty lines is often not needed. Empty lines is a tool to communicate/separate slightly more medium-size blocks of logically related code.

E.g. here I would recommend making it one block:

```golang
var start, end roachpb.Key
if parts[0] != "X" {
	start = roachpb.Key(parts[0])
}
if parts[1] != "X" {
	end = roachpb.Key(parts[1])
}
```

Also, could make it more concise like:

```golang
var span roachpb.Span
if parts[0] != "X" {
	span.Key = roachpb.Key(parts[0])
}
if parts[1] != "X" {
	span.EndKey = roachpb.Key(parts[1])
}
return span
```